woocommerce/woocommerce
# Add coming soon template tracks and update block attributes

### Changes proposed in this Pull Request:

Closes #53074

The best available way I found to track which coming soon pattern is used is by hooking on template save, and identifying the pattern based on an artificial ID we inject in the pattern.

This PR:

- Remove `storeOnly` and add `comingSoonPatternId` attribute for new templates
- Add `comingSoonPatternId` attribute to all coming soon patterns
- Set the default pattern to `woocommerce/page-coming-soon-default`
- Add tracks on template save to track which coming soon pattern is chosen



### How to test the changes in this Pull Request:

Note: If you're testing on a non-fresh site, you'll need to disable or delete the [pattern cache](https://github.com/woocommerce/woocommerce/blob/8b71979dc980675a8f2bd72b52bd5c55889293fe/plugins/woocommerce/src/Blocks/BlockPatterns.php#L127-L131).

#### Testing tracks

1. Ensure WCA Test Helper plugin is installed and activated
2. Enable `coming-soon-newsletter-template` feature flag
3. Go through the core profiler and ensure tracking is enabled
4. Go to `Appearance > Editor > Templates`, look for `Page: Coming soon` and open it
5. Open the settings sidebar on the right side of the screen
6. Open the `Design` panel and select any of the patterns available, ex: `Coming Soon Minimal Left Image`
7. Click on `Save`
8. Go to `WooCommerce > Status > Logs`
9. Open the latest `tracks` log
10. Look for the track `wcadmin_coming_soon_template_saved`, it should have the prop `pattern_id` with the value being the ID of the selected pattern, ex: `page-coming-soon-minimal-left-image`
11. Test multiple patterns and ensure all of them triggers the track with its own unique ID

<img width="920" alt="image" src="https://github.com/user-attachments/assets/f618d2c6-d08f-458f-8a8c-67eb9569da51" />

#### Testing new method to set background color

1. Go to `Appearance > Editor > Templates` and look for `Page: Coming soon` and open it
2. Open the settings sidebar on the right side of the screen
6. Open the `Design` panel and select any of the patterns available, ex: `Coming Soon Minimal Left Image`
1. Open the `Document overview` (the stack of bars icon, top left)
5. Click on `Coming soon` block
6. On the settings sidebar, click on the `Block` tab
7. Observe you see the `Background` option and it's pre-populated with a color
8. On any patterns except for `Coming Soon Modern Black` where the background image includes color that covers the page, try changing background
9. Observe the page's background change
10. Select other patterns and test changing its background

<img width="1407" alt="image" src="https://github.com/user-attachments/assets/2a78f0ca-b326-42f9-acc9-d92a96ce97c3" />

#### Regression test on existing coming soon pages

1. Create a fresh JN with WooCommerce 9.5.1 ([link](https://jurassic.ninja/create?woocommerce-beta-tester&woocommerce&wp-debug-log&nojetpack))
1. Go to `Appearance > Editor > Templates` and look for `Page: Coming soon` and open it
2. Make some modifications on the page and save
3. Go to `WooCommerce > Live branches` and activate the branch `update/53074-coming-soon-template-tracks`
4. Go to `Tools > WCA Test Helper > Features`
5. Enable `coming-soon-newsletter-template` feature flag
6. Go back to the coming soon template
7. Observe the template does not have a "Attempt block recovery" notice
2. Open the settings sidebar on the right side of the screen
1. Open the `Document overview` (the stack of bars icon, top left)
5. Click on `Coming soon` block
6. On the settings sidebar, click on the `Block` tab
7. Observe there's `Settings` tab with the cogs icon
8. Observe you can change the background color using the color picker

<img width="719" alt="image" src="https://github.com/user-attachments/assets/5d57ee92-cc5b-4481-9cff-ca559703cd22" />

#### Optional: Regression test on feature flag disabled

1. Go to `Tools > WCA Test Helper > Features`
1. Disable `coming-soon-newsletter-template` feature flag
1. Open the coming soon template site editor
1. On the right sidebar, open up `Template` tab
2. Observe no `Design` panel available
3. Reset the template and refresh
4. Observe the coming soon page is rendered properly with no background color glitches



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
